cpp/src/arrow/scalar_test.cc

```diff
@@ -866,6 +866,9 @@ TEST(TestTimeScalars, Basics) {
     ASSERT_TRUE(first->Equals(*MakeScalar(ty, 5).ValueOrDie()));
     ASSERT_TRUE(last->Equals(*MakeScalar(ty, 42).ValueOrDie()));
     ASSERT_FALSE(last->Equals(*MakeScalar("string")));
+
+    ASSERT_OK_AND_ASSIGN(auto casted, first->CastTo(ty));
+    ASSERT_TRUE(casted->Equals(*first));
```

Comment:
Can we create a generic test for identity cast instead? Something like:
```c++
TEST(TestIdentityCast, AllTypes) {
  RandomArrayGenerator rng(/*seed=*/42);
  for (auto type : {...}) {
    ASSERT_OK_AND_ASSIGN(
        auto scalar, rng.ArrayOf(type, /*size=*/1).GetScalar(0));
    ASSERT_OK_AND_ASSIGN(auto casted, scalar->CastTo(ty));
    ASSERT_TRUE(casted->Equals(*scalar));
  }
}
```

Reply:
done